packages/core/src/machines/actors/auth/resetPassword.ts

```diff
@@ -0,0 +1,115 @@
+import { createMachine, assign, sendUpdate } from 'xstate';
+
+import { AuthEvent, ResetPasswordContext } from '../../../types';
+import { Auth } from 'aws-amplify';
+
+export const resetPasswordActor = createMachine<
+  ResetPasswordContext,
+  AuthEvent
+>(
+  {
+    id: 'resetPasswordActor',
+    initial: 'resetPassword',
+    states: {
+      resetPassword: {
+        initial: 'edit',
+        exit: ['clearFormValues', 'clearError'],
+        states: {
+          edit: {
+            entry: sendUpdate(),
+            on: {
+              SUBMIT: 'submit',
+              CHANGE: { actions: 'handleInput' },
+            },
+          },
+          submit: {
+            entry: [sendUpdate(), 'setUsername', 'clearError'],
+            invoke: {
+              src: 'resetPassword',
+              onDone: {
+                target: '#resetPasswordActor.confirmResetPassword',
+              },
+              onError: {
+                actions: ['setRemoteError'],
+                target: 'edit',
+              },
+            },
+          },
+        },
+      },
+      confirmResetPassword: {
+        initial: 'edit',
+        exit: ['clearFormValues', 'clearError', 'clearUsername'],
+        states: {
+          edit: {
+            entry: sendUpdate(),
+            on: {
+              SUBMIT: 'submit',
+              RESEND: 'resendCode',
+              CHANGE: { actions: 'handleInput' },
+            },
+          },
+          resendCode: {
+            entry: ['clearError', sendUpdate()],
+            invoke: {
+              src: 'resetPassword',
+              onDone: { target: 'edit' },
+              onError: {
+                actions: 'setRemoteError',
+                target: 'edit',
+              },
+            },
+          },
+          submit: {
+            entry: ['clearError', sendUpdate()],
+            invoke: {
+              src: 'confirmResetPassword',
+              onDone: {
+                actions: 'clearUsername',
+                target: '#resetPasswordActor.resolved',
+              },
+              onError: {
+                actions: 'setRemoteError',
+                target: 'edit',
+              },
+            },
+          },
+        },
+      },
+      resolved: { type: 'final' },
+    },
+  },
+  {
+    actions: {
+      setRemoteError: assign({
+        remoteError: (_, event) => event.data?.message || event.data,
+      }),
+      setUsername: assign({
+        username: (context) => context.formValues.username,
+      }),
+      handleInput: assign({
+        formValues: (context, event) => {
+          const { name, value } = event.data;
+          return { ...context.formValues, [name]: value };
+        },
+      }),
+      clearFormValues: assign({ formValues: {} }),
+      clearError: assign({ remoteError: '' }),
+      clearUsername: assign({ username: undefined }),
+    },
+    services: {
+      async resetPassword(context) {
+        const { username } = context.formValues;
```

Comment:
`username` won't always be in the `context.formValues`. For instance, we call this method when we `RESEND` the code on the `ConfirmResetPassword` screen, but there is no `username` field on that screen, so it gets sent up as a `data` property on the `onClick` call.

Reply:
Good catch -- I'll do `context.formValues?.username ?? context.username`.

Reply:
Done here https://github.com/aws-amplify/amplify-ui/pull/180/commits/d67bb44bafa8d2f7e7396cb096eeb1180eed0e9a